node/internal.ts

```diff
@@ -426,7 +427,7 @@ export function _which(tool: string, check?: boolean): string {
 function _tryGetExecutablePath(filePath: string, extensions: string[]): string {
     try {
         // test file exists
-        let stats: fs.Stats = fs.statSync(filePath);
+        let stats: fs.Stats = fs.statSync(_unTildify(filePath));
```

Comment:
filePath in tryGetExe is the segment of the PATH env variable which is what would be prefixed with a tilde.

It's fs.statSync that fails if there is a tilde. See the Node issue where they won't fix

So untildify just expands tilde only when it's a prefix as that should only be the case

All the Tests pass which include tests for running ls, grep, etc